cdc/processor.go

```diff
@@ -539,35 +552,53 @@ func (p *processor) syncResolved(ctx context.Context) error {
 				return errors.Trace(err)
 			}
 			if !t.IsDDL() {
+				log.Warn("Receive non-DDL txn from ddlJobsCh", zap.Uint64("ts", t.Ts))
 				continue
 			}
 			p.schemaStorage.AddJob(t.DDL.Job)
+			if err := flush(ctx); err != nil {
+				return errors.Trace(err)
+			}
 			atomic.StoreUint64(&p.ddlResolveTS, rawTxn.Ts)
-		case e, ok := <-p.resolvedTxns:
+		case rawTxn, ok := <-p.resolvedTxns:
 			if !ok {
 				return nil
 			}
-			if e.IsResolved {
+			if rawTxn.IsResolved {
+				// TODO: Avoid flushing for every resolved message
+				if err := flush(ctx); err != nil {
+					return errors.Trace(err)
+				}
 				select {
-				case p.executedTxns <- e:
+				case p.executedTxns <- rawTxn:
+					continue
 				case <-ctx.Done():
 					return errors.Trace(ctx.Err())
 				}
-			} else {
-				err := p.schemaStorage.HandlePreviousDDLJobIfNeed(e.Ts)
-				if err != nil {
-					return errors.Trace(err)
-				}
-				txn, err := p.mounter.Mount(e)
-				if err != nil {
-					return errors.Trace(err)
-				}
-				if err := p.sink.Emit(ctx, txn); err != nil {
+			}
+
+			if err := p.schemaStorage.HandlePreviousDDLJobIfNeed(rawTxn.Ts); err != nil {
+				return errors.Trace(err)
+			}
+			txn, err := p.mounter.Mount(rawTxn)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			pendingTxns = append(pendingTxns, txn)
```

Comment:
I think we should filter the fake txn here.

Reply:
Isn't fake txns already filtered in the `ddlJobsCh` case?